lib/vrl/stdlib/benches/benches.rs

```diff
@@ -1418,6 +1418,11 @@ bench_function! {
         args: func_args![value: r#"{"key": "value"}"#],
         want: Ok(value!({key: "value"})),
     }
+
+    map_max_depth {
+        args: func_args![value: r#"{"key": "value"}"#, max_depth: 128],
+        want: Ok(value!({key: "value"})),
+    }
```

Comment:
Could you add another benchmark, that actually iterates over, say 5 nested objects/arrays?  Because right now, it does look like the performance is 2,5x worse, without it actually doing any recursion, so I'm curious what happens to those numbers if we _do_ let it recurse.